src/main/scala/org/broadinstitute/dsde/firecloud/model/MethodRepository.scala

```diff
@@ -25,12 +25,41 @@ object MethodRepository {
     name: Option[String] = None,
     snapshotId: Option[Int] = None,
     synopsis: Option[String] = None,
-    owner: Option[String] = None,
+    documentation: Option[String] = None,
     createDate: Option[String] = None,
     url: Option[String] = None,
+    payload: Option[String] = None,
     entityType: Option[String] = None
   )
 
+  case class MethodId(
+    namespace: String,
+    name: String,
+    snapshotId: Int
+  )
+
+  case class MethodCreate(
+    namespace: String,
+    name: String,
+    synopsis: String,
+    documentation: String,
+    payload: String,
+    entityType: String
+  )
+
+  case class EditMethodRequest(
+    source: MethodId,
+    synopsis: String,
+    documentation: String,
+    payload: String,
+    redactOldSnapshot: Boolean
+  )
+
+  case class EditMethodResponse(
+    method: Method,
+    message: Option[String] = None
+  )
```

Comment:
do we really need all these new model classes? For instance, since the `Method` class uses `Options`, we could possibly replace `MethodCreate` and `MethodId` with `Method`. This is a brainstorm, not an explicit directive.